cvat/apps/iam/permissions.py

```diff
@@ -100,6 +106,22 @@ class OpenPolicyAgentPermission(metaclass=ABCMeta):
     scope: str
     obj: Optional[Any]
 
+    @classmethod
+    @abstractmethod
+    def _get_scopes(cls, request: ExtendedRequest, view: ViewSet, obj: Any) -> list:
+        """Method to override to define scopes based on the request"""
+
+    @classmethod
+    def get_scopes(cls, request: ExtendedRequest, view: ViewSet, obj: Any):
+        with suppress(KeyError):
```

Comment:
The problem with this approach is that it will also suppress the exception when a view _does_ support the method, but the corresponding permission class doesn't define a scope for it. In this case, the code is broken, so we _should_ propagate the exception (and consequently return a 500 error).

Do you think you could implement this behavior?

Reply:
I think so